File: backend/Entities/Models/Pessoa.cs
using System;
using System.Collections.Generic;

namespace Entities.Models;

public partial class Pessoa
{
    public Guid Id { get; set; }

    public string Nome { get; set; } = null!;

    public string Fantasia { get; set; } = null!;

    public DateTime DataCadastro { get; set; }

    public virtual ICollection<Documento> DocumentoDestinatarios { get; } = new List<Documento>();

    public virtual ICollection<Documento> DocumentoEmitentes { get; } = new List<Documento>();

    public virtual ICollection<Documento> DocumentoRemetentes { get; } = new List<Documento>();

    public virtual ICollection<PessoaEndereco> PessoaEnderecos { get; } = new List<PessoaEndereco>();

    public virtual PessoaFisica? PessoaFisica { get; set; }

    public virtual PessoaJuridica? PessoaJuridica { get; set; }

    public virtual PessoaOutro? PessoaOutro { get; set; }

    public virtual ICollection<Produto> Produtos { get; } = new List<Produto>();

    public virtual Transportador? Transportador { get; set; }

    public virtual Transportador1? Transportador1 { get; set; }

    public virtual ICollection<Veiculo> Veiculos { get; } = new List<Veiculo>();
}

File: frontend/src/app/Business/Models/pessoa.ts
import { Documento } from "./dcumento";
import { PessoaEndereco } from "./pessoaEndereco";
import { PessoaFisica } from "./pessoaFisica";
import { PessoaJuridica } from "./pessoaJuridica";
import { PessoaOutro } from "./pessoaOutro";
import { Produto } from "./produto";
import { Transportador } from "./transportador";
import { Transportador1 } from "./transportador1";
import { Veiculo } from "./veiculo";


export interface Pessoa  {
    dataCadastro: string;
    documentoDestinatarios: Documento[];
    documentoEmitentes: Documento[];
    documentoRemetentes: Documento[];
    fantasia: string;
    id: string;
    nome: string;
    pessoaEnderecos: PessoaEndereco [];
    pessoaFisica: PessoaFisica | null;
    pessoaJuridica: PessoaJuridica | null;
    pessoaOutro: PessoaOutro | null;
    produtos: Produto[];
    transportador: Transportador | null;
    transportador1: Transportador1 | null;
    veiculos: Veiculo[];
}